src/Dependencies/Collections/ImmutableSegmentedHashSet`1.cs

```diff
@@ -10,6 +10,57 @@
 
 namespace Microsoft.CodeAnalysis.Collections
 {
+    /// <summary>
+    /// Represents a segmented hash set that is immutable; meaning it cannot be changed once it is created.
+    /// </summary>
+    /// <remarks>
+    /// <para>There are different scenarios best for <see cref="ImmutableSegmentedHashSet{T}"/> and others
+    /// best for <see cref="ImmutableHashSet{T}"/>.</para>
+    ///
+    /// <para>The following table summarizes the performance characteristics of
+    /// <see cref="ImmutableSegmentedHashSet{T}"/>:</para>
+    /// 
+    /// <list type="table">
```

Comment:
As an aside, you have a lot of great context on these data structures. When I joined the team, I would have *loved* to have a document that basically outlines how each of them essentially works, their pitfalls and strengths, and their basic performance characteristics. Even after having been on the team for nearly a year, I still think I would find such a document useful.